jwst/assign_wcs/nirspec.py

```diff
@@ -726,8 +726,7 @@ def get_open_msa_slits(prog_id, msa_file, msa_metadata_id, dither_position,
             quadrant = slitlet_rows[0]['shutter_quadrant']
             ycen = j
             xcen = slitlet_rows[0]['shutter_row']  # grab the first as they are all the same
-            shutter_id = xcen + (ycen - 1) * 365  # shutter numbers in MSA file are 1-indexed
-
+            shutter_id = np.int64(xcen) + (np.int64(ycen) - 1) * 365  # shutter numbers in MSA file are 1-indexed
```

Comment:
Is it necessary to use numpy scalars? The only reason we end up with numpy scalars is the indexing of FITS_rec array.  It may be better to use `.item()` as in 

`xcen = slitlet_rows[0]['shutter_row'].item()`

Reply:
Testing locally, changing that line to use .item() doesn't fix the integer overflow problem in the shutter_id calculation under numpy 2, but Robert's changes do.  I think it needs the explicit upcast.

Reply:
Actually, if the .item() attribute of both xcen and ycen is used, the test passes.  I think it's a matter of taste which to use; personally, I prefer the numpy casting, as it works on both numpy scalars and python scalars, whereas the .item() method only works for numpy arrays or scalars, not for Python scalars.

Reply:
The physical meaning of xcen and ycen is an ID so to me a Python scalar makes sense. However, there's no harm if it's a numpy scalar, the extra stuff is just not used. Approving the PR.